app/javascript/components/SignIn.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import Axios from 'axios';
+
+export default class SigInForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { user: {email: '', password: ''} };
+    this.handleSignIn = this.handleSignIn.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleSignIn = (e) => {
+    e.preventDefault();
+    Axios({
+      method: 'post',
+      url: '/auth/sign_in',
+      data: this.state.user
+    }).then(response => {
+      sessionStorage.setItem('user',
+      JSON.stringify({
+        'access-token': response.request.getResponseHeader('access-token'),
+        'token-type': response.request.getResponseHeader('token-type'),
+        client: response.headers["client"],
+        expiry: response.headers["expiry"],
+        uid: response.headers["uid"]
+      }));
+      this.props.history.push('/');
+      }).catch(error => {
+      this.setState({ errors: error.response.data.errors })
+      })
+  };
+
+  handleChange = (user) => {
+    const currentUser = this.state.user;
+    currentUser[user.target.name] = user.target.value;
+    this.setState(currentUser);
+  }
+
+  render() {
+    let message;
+    if (this.state.errors) {
+      message = <div>
+        {this.state.errors.map((error) => {
+          return(
+              <p>{error}</p>
+          )
+        })}
+      </div>
+    }
+    return (
+      <div>
+        <div>
```

Comment:
For better readability I'd add className like 'errors'
Also it is worth choosing better name for variable (e.g. errorMessages)